src/module-config/module-config.test.ts

```diff
@@ -118,6 +119,101 @@ describe("getConfig", () => {
     await expect(fooConfig).resolves.toHaveProperty("foo", "qux");
     const barConfig = Config.getConfig("bar-module");
     await expect(barConfig).resolves.toHaveProperty("bar", "baz");
+
+    // TEST CASE 2: One that bugs out for multiple modules: see conversation on:
+    // https://github.com/openmrs/openmrs-esm-module-config/pull/18
+    Config.clearAll();
+    Config.defineConfigSchema("@openmrs/esm-patient-chart", {
```

Comment:
This is a very complex example of the problem. It doesn't make it clear what exactly the issue is. Could you please reduce it to a [minimal working example](https://stackoverflow.com/help/minimal-reproducible-example)?